### PR TITLE
Replace webpack with esbuild

### PR DESCRIPTION
**Pull request by fregante**
- Described in https://github.com/refined-github/refined-github/issues/4008#issuecomment-964072533

To do:

- [ ] JSX pragma: dom-chef
- [ ] Test at all 
- [ ] Add "build done" output after every compilation, related: https://github.com/evanw/esbuild/issues/884#issuecomment-973926939

Notes:

- It shows a warning: https://github.com/evanw/esbuild/issues/1785

### CODE REVIEW

File: build/esbuild.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {createRequire} from 'node:module';
+
+import esbuild from 'esbuild';
+
+import {getFeatures, getFeaturesMeta} from './readme-parser.js';
+
+const {resolve: resolvePackage} = createRequire(import.meta.url);
```
Comment (fregante) [inline build/esbuild.ts:7]: Space generously gifted by our lint rules 😌

**Review by fregante — COMMENTED** [review]
Untested

Screen Shot 2

**Comment by fregante** [comment]
Blocked by 

- https://github.com/evanw/esbuild/issues/1492

🤷‍♂️

**Comment by hyrious** [comment]
I'll give you 2 plugins to solve `import.meta.url` and the warning of `typed-query-selector`:

```js
const importMeta: esbuild.Plugin = {
	name: 'Import Meta',
	setup(build) {
		build.onLoad({filter: /source.features.\S+\.tsx$/}, async args => {
			let contents = await readFile(args.path, 'utf8');
			contents = contents.replace(
				/\bimport\.meta\.url\b/g,
				JSON.stringify(pathToFileURL(args.path)),
			);
			return { contents, loader: 'default' }
		});
	},
};

const TypedQuerySelector: esbuild.Plugin = {
	name: 'typed-query-selector',
	setup(build) {
		build.onResolve({filter: /^typed-query-selector$/}, ({path}) => {
			return { path, namespace: 'typed-query-selector' };
		});
		build.onLoad({filter: /()/, namespace: 'typed-query-selector'}, (a) => {
			return { contents: '', loader: 'js' };
		});
	},
}
```

**Comment by fregante** [comment]
Thank you!

**Pull request closed without merging**